### PR TITLE
gas: add gas fee config for different tx types

### PR DESCRIPTION
**Pull request by forcodedancing**
### Description

Add gas fee config for different transaction types, and check `gas amount` when sending transactions.

### Rationale

1) The costs of different transaction types are different. 
2) There should be check of transactions' gas amount.

### Example

NA

### Changes

Notable changes:
* remove `getWithdrawGasFee` api

### CODE REVIEW

File: core/executor/executor.go [hunk 1]
```diff
@@ -12,15 +13,15 @@ import (
...
+	VerifyGas(gasAccountIndex, gasFeeAssetId int64, txType int, gasFeeAmount *big.Int, skipGasAmtChk bool) error
 	StateDB() *sdb.StateDB
 	DB() *sdb.ChainDB
 	CurrentBlock() *block.Block
 }
 
 type TxExecutor interface {
 	Prepare() error
-	VerifyInputs() error
+	VerifyInputs(bool) error
```
Comment (yutianwu) [inline core/executor/executor.go:24]: can we have a name for the param, it's not obvious at the first glance

Comment (forcodedancing) [reply core/executor/executor.go]: sure, updated

File: core/executor/mint_nft_executor.go [hunk 2]
```diff
@@ -10,6 +10,7 @@ import (
 
 	"github.com/bnb-chain/zkbnb-crypto/ffmath"
 	"github.com/bnb-chain/zkbnb-crypto/wasm/txtypes"
+
```
Comment (yutianwu) [inline core/executor/mint_nft_executor.go:13]: fix the imports

File: tools/dbinitializer/main.go [hunk 3]
```diff
@@ -122,10 +122,31 @@ func Initialize(
 }
 
 func initSysConfig(svrConf *contractAddr, bscTestNetworkRPC, localTestNetworkRPC string) []*sysconfig.SysConfig {
+
+	// to config gas for different transaction types, need to be evaluated and tune these values
+	gasFeeConfig := make(map[uint32]map[int]int64) // asset id -> (tx type -> gas fee value)
+	gasFeeConfig[0] = make(map[int]int64)          // bnb asset
+	gasFeeConfig[0][types.TxTypeTransfer] = 10000000000000
```
Comment (yutianwu) [inline tools/dbinitializer/main.go:129]: can we new initiate a map first and then assign it to `gasFeeConfig`, and can we avoid using the magic `0`.

Comment (forcodedancing) [reply tools/dbinitializer/main.go]: sure, updated

File: core/executor/mint_nft_executor.go (hunk 2 above)
Comment (forcodedancing) [reply core/executor/mint_nft_executor.go]: updated

**Review by yutianwu — APPROVED** [review]

**Review by KeefeL — APPROVED** [review]
lgtm

**Pull request merged**